crates/crypto/bls/src/supranational/pubkey.rs

```diff
@@ -16,3 +16,13 @@ impl PubKey {
         BlstPublicKey::from_bytes(&self.inner).map_err(|err| BLSError::BlstError(err.into()))
     }
 }
+
+impl Validate for PubKey {
+    type Error = BLSError;
+
+    fn validate(&self) -> std::result::Result<(), errors::BLSError> {
+        self.to_blst_pubkey()?
+            .validate()
+            .map_err(|err| BLSError::BlstError(err.into())) // Convert `BLST_ERROR` to `BLSError`
```

Comment:
```suggestion
            .validate()
             // Convert `BLST_ERROR` to `BLSError`
            .map_err(|err| BLSError::BlstError(err.into()))
```

Doing errors like this makes it a bit more clear